src/de/jost_net/JVerein/gui/dialogs/BuchungenMitgliedskontenZuordnungDialog.java

```diff
@@ -0,0 +1,535 @@
+    {
+
+      @Override
+      public void handleAction(Object context)
+      {
+        throw new OperationCanceledException();
+      }
+    }, null, false, "stop-circle.png");
+    group.addButtonArea(buttons);
+    getShell()
+        .setMinimumSize(getShell().computeSize(WINDOW_WIDTH, SWT.DEFAULT));
+  }
+
+  /**
+   * Ermittelt eine Liste von Buchungen mit Mitgliedskonten und gibt
+   * diese an den BuchungenMitgliedskontenZuordnungVorschauDialog weiter
+   * 
+   */
+  private void doSearchAssignment()
+  {
+    final Date dateFromInput = (Date)dateFrom.getValue();
+    final Date dateUntilInput = (Date)dateUntil.getValue();
+
+    final boolean useIbanInput = Boolean.TRUE.equals(this.useIban.getValue());
+    final boolean useMemberNumberInput = Boolean.TRUE.equals(this.useMemberNumber.getValue());
+    final boolean useNameInput = Boolean.TRUE.equals(this.useName.getValue());
+
+    settings.setAttribute(SETTINGS_NAME_IBAN, useIbanInput);
+    settings.setAttribute(SETTINGS_NAME_MITGLIEDSNUMMER, useMemberNumberInput);
+    settings.setAttribute(SETTINGS_NAME_VORNAME_NAME, useNameInput);
+
+    BackgroundTask t = new BackgroundTask()
+    {
+
+      @Override
+      public void run(ProgressMonitor monitor) throws ApplicationException
+      {
+        try 
+        {
+          boolean externeMitgliedsnummer = Boolean.TRUE.equals(Einstellungen.getEinstellung().getExterneMitgliedsnummer());
+
+          if(!useIbanInput && !useMemberNumberInput && !useNameInput)
+          {
+            GUI.getStatusBar().setErrorText("Es wurde keine Zuordnungsart angegeben.");
+            return;
+          }
+
+          if(dateFromInput == null || dateUntilInput == null)
+          {
+            GUI.getStatusBar().setErrorText("Bitte geben Sie ein Start- und Enddatum ein.");
+            return;
+          }
+
+          if(dateUntilInput.before(dateFromInput))
+          {
+            GUI.getStatusBar().setErrorText("Das Enddatum liegt vor dem Startdatum.");
+            return;
+          }
+
+          Map<String, String> uniqueIbans = new HashMap<>();
+          Set<String> duplicateIbans = new HashSet<>();
+
+          Map<String, String> uniqueIds = new HashMap<>();
+          Set<String> duplicateIds = new HashSet<>();
+
+          Map<String, String> uniqueNames = new HashMap<>();
+          Set<String> duplicateNames = new HashSet<>();
+
+          if(useIbanInput || useMemberNumberInput || useNameInput)
+          {
+            DBIterator<Mitglied> mitglieder = Einstellungen.getDBService().createList(Mitglied.class);
+            mitglieder.addFilter("((eintritt <= ? and (austritt is null or austritt >= ?)) "
+                             + "or (eintritt <= ? and (austritt is null or austritt >= ?)) "
+                             + "or (eintritt > ? and austritt < ?))",
+                dateFromInput, dateFromInput, 
+                dateUntilInput, dateUntilInput,
+                dateFromInput, dateUntilInput
+            );
```

Comment:
Ich impliziere, dass dateFromInput <= dateUntilInput ist. Dann würde ich, um alle aktiven Mitglieder zu erhalten, den Filter ersetzen gegen:
`mitglieder.addFilter("austritt IS NULL OR austritt > ?", dateUntilInput);`